starknet/tests/eth-sig-auth.test.ts

```diff
@@ -0,0 +1,258 @@
+      contractAddress: ethSigAuthAddress,
+      entrypoint: 'authenticate_propose',
+      calldata: CallData.compile(proposeCalldata as any),
+    });
+
+    console.log(result);
+
+    // UPDATE PROPOSAL
+
+    const updateProposalMsg: UpdateProposal = {
+      authenticator: ethSigAuthAddress,
+      space: spaceAddress,
+      author: signer.address,
+      proposalId: '0x1',
+      executionStrategy: {
+        address: '0x0000000000000000000000000000000000005678',
+        params: ['0x0'],
+      },
+      metadataURI: ['0x1', '0x2', '0x3', '0x4'],
+      salt: '0x7',
+    };
+
+    sig = await signer.signTypedData(domain, updateProposalTypes, updateProposalMsg);
+    splitSig = getRSVFromSig(sig);
+
+    const updateProposalCalldata = {
+      r: cairo.uint256(splitSig.r),
+      s: cairo.uint256(splitSig.s),
+      v: splitSig.v,
+      space: updateProposalMsg.space,
+      author: updateProposalMsg.author,
+      proposalId: cairo.uint256(updateProposalMsg.proposalId),
+      executionStrategy: updateProposalMsg.executionStrategy,
+      metadataURI: updateProposalMsg.metadataURI,
+      salt: cairo.uint256(updateProposalMsg.salt),
+    };
+
+    result = await account0.execute({
+      contractAddress: ethSigAuthAddress,
+      entrypoint: 'authenticate_update_proposal',
+      calldata: CallData.compile(updateProposalCalldata as any),
+    });
+
+    console.log(result);
+
+    {
+      // Random Tx just to advance the block number on the devnet so the voting period begins.
+
+      await account0.declareAndDeploy({
+        contract: json.parse(
+          fs
+            .readFileSync('starknet/target/dev/sx_EthSigAuthenticator.sierra.json')
+            .toString('ascii'),
+        ),
+        casm: json.parse(
+          fs.readFileSync('starknet/target/dev/sx_EthSigAuthenticator.casm.json').toString('ascii'),
+        ),
+        constructorCalldata: CallData.compile({}),
+      });
+    }
+
+    // VOTE
+
+    const voteMsg: Vote = {
+      authenticator: ethSigAuthAddress,
+      space: spaceAddress,
+      voter: signer.address,
+      proposalId: '0x1',
+      choice: '0x1',
+      userVotingStrategies: [{ index: '0x0', params: ['0x1', '0x2', '0x3', '0x4'] }],
+      metadataURI: ['0x1', '0x2', '0x3', '0x4'],
+    };
+
+    sig = await signer.signTypedData(domain, voteTypes, voteMsg);
+    splitSig = getRSVFromSig(sig);
+
+    const voteCalldata = {
+      r: cairo.uint256(splitSig.r),
+      s: cairo.uint256(splitSig.s),
+      v: splitSig.v,
+      space: voteMsg.space,
+      voter: voteMsg.voter,
+      proposalId: cairo.uint256(voteMsg.proposalId),
+      choice: voteMsg.choice,
+      userVotingStrategies: voteMsg.userVotingStrategies,
+      metadataURI: voteMsg.metadataURI,
+    };
+
+    result = await account0.execute({
+      contractAddress: ethSigAuthAddress,
+      entrypoint: 'authenticate_vote',
+      calldata: CallData.compile(voteCalldata as any),
+    });
+
+    console.log(result);
```

Comment:
could we assert that the vote was counted correctly? just in case for some reason it didn't revert but didn't count the vote as well